### PR TITLE
Create a common scrape class

### LINKED ISSUE #60: Create common HTTP class for scraping / crawling
We have Crawl and Scrape workers which have some common functionality when it comes to actually fetching a URL and collecting its html.  Rather than implement things like HTTP error handling in two places we should create something like an AbstractScraper which covers all that, and have the AbstractCrawler and AbstractClaimScraper both extend it.

The AbstractScraper should:

1. Gracefully handle HTTP errors

I guess that's it for now...

This is an enhancement of the work completed for #20

### PR DESCRIPTION
**Pull request by slifty**
Statement scrapes and crawls are both different types of scraper.  As described in #60, there is value in having both types of scraper extend a common class so we can avoid redundant supporting logic.

This PR implements that common class.  It also sneaks in a little bit of augmentation to our logger (and a bug fix where stack trace would render as undefined when a log line didn't include an error)

This resolves #60

### CODE REVIEW

File: src/server/workers/scrapers/AbstractScraper.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  // (this is an abstract method and we need to define its footprint.)
+  // eslint-disable-next-line no-unused-vars
+  scrapeHandler = (responseString) => {
+    throw new Error('You implemented a scraper but forgot to define the scrapeHandler.')
+  }
+
+  /**
+   * Runs the scraper
+   *
+   * @return {Promise} a Promise to provide the results of the scrape
```
Comment (reefdog) [inline src/server/workers/scrapers/AbstractScraper.js]: `a Promise` → `A Promise`

File: src/server/workers/scrapers/AbstractStatementScraper.js [hunk 2]
```diff
@@ -1,40 +1,23 @@
...
    *
-   * Each scraper that extends AbstractStatementScraper needs to implement its own
-   * scrapeHandler method.
+   * Each statement scraper that extends AbstractStatementScraper needs to implement
+   * its own statementScrapeHandler method.
    *
-   * @param {String} htmlString The HTML or JSON that came from the HTTP request
-   * @return {Object[]}         the list of statements that were scraped
+   * @param {String} responseString The HTML or JSON that came from the HTTP request
+   * @return {Object[]}             the list of statements that were scraped
```
Comment (reefdog) [inline src/server/workers/scrapers/AbstractStatementScraper.js]: `the list` → `The list`

File: src/server/config.js [hunk 3]
```diff
@@ -4,9 +4,11 @@ dotenv.config()
 
 const {
   ENABLED_QUEUE_DIRECTORIES,
+  LOG_LEVEL,
 } = process.env
 
 export default {
   ...process.env,
   ENABLED_QUEUE_DIRECTORIES: ENABLED_QUEUE_DIRECTORIES.split(','),
+  LOG_LEVEL: LOG_LEVEL || 'info',
```
[the remark this answers is not in the document]
Comment (slifty) [reply src/server/config.js]: I agree, it's perfect as it is.

(This joke doesn't get old IMHO)

[the remark this answers is not in the document]
Comment (slifty) [reply src/server/config.js]: Waht do you think of just setting the default in the logger.js (that is where it was set originally) *and* add it to .env.example

[the remark this answers is not in the document]
Comment (reefdog) [reply src/server/config.js]: That seems sensible.

[the remark this answers is not in the document]
Comment (reefdog) [reply src/server/config.js]: pleasedwithmyself

**Pull request merged**